### DISCUSSION TITLE
Passing i18next instance i Remix stub?

### DISCUSSION THREAD

**Discussion by stonetwig** [sergiodxa/remix-i18next#218] [Q&A] [answered]
Hello,

I am having some issues with my tests after introducing remix-18next which is a great library.

I get the following error:
`react-i18next:: You will need to pass in an i18next instance by using initReactI18next`

I am not sure how to parse this information though. Not sure on how to proceed?
I have googled a lot and searched in this repo but haven't found anything of value.

Most tests aren't e2e tests, so in this case there isn't any backend?

I have done this:
```
render(<RemixStub hydrationData={{
      loaderData: { root: { locale: "cimode" } },
    }} />);
```

But it had no effect. Any help would be greatly appreciated!

**Comment by stonetwig** [comment, marked as answer]
Solved it by mocking it in setup-test-env.ts

```
vi.mock('react-i18next', () => ({
  // this mock makes sure any components using the translate hook can use it without a warning being shown
  useTranslation: () => {
    return {
      t: (str: string) => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
  initReactI18next: {
    type: '3rdParty',
    init: () => {},
  },
}));
```

**Comment by sergiodxa** [comment]
You need to wrap RemixStub in a IntlProvider by react-i18next and pass an i18next instance.
